t/python/t0013-job-list.py

```diff
@@ -74,170 +74,239 @@ def waitForConsistency(self, jobs_list_length):
         )
 
         jobs = self.getJobs(rpc_handle)
 
-        rpc_handle = flux.job.job_list_inactive(
-            self.fh, jobs[4]["t_inactive"], 10, self.attrs
-        )
+        rpc_handle = flux.job.job_list_inactive(self.fh, jobs[4]["t_inactive"], 10)
 
         jobs_inactive = self.getJobs(rpc_handle)
 
         self.assertEqual(len(jobs_inactive), 4)
 
     # flux job list-inactive with middle timestamp #1
-    def test_08_middle_timestamp_1(self):
+    def test_13_list_inactive_middle_timestamp_1(self):
         rpc_handle = flux.job.job_list(
             self.fh, 20, ["t_inactive"], states=flux.constants.FLUX_JOB_STATE_INACTIVE
         )
 
         jobs = self.getJobs(rpc_handle)
 
-        rpc_handle = flux.job.job_list_inactive(
-            self.fh, jobs[5]["t_inactive"], 20, self.attrs
-        )
+        rpc_handle = flux.job.job_list_inactive(self.fh, jobs[5]["t_inactive"], 20)
 
         jobs_inactive = self.getJobs(rpc_handle)
 
         self.assertEqual(len(jobs_inactive), 5)
 
     # flux job list-inactive with middle timestamp #2
-    def test_09_middle_timestamp_2(self):
+    def test_14_list_inactive_middle_timestamp_2(self):
         rpc_handle = flux.job.job_list(
             self.fh, 20, ["t_inactive"], states=flux.constants.FLUX_JOB_STATE_INACTIVE
         )
 
         jobs = self.getJobs(rpc_handle)
 
-        rpc_handle = flux.job.job_list_inactive(
-            self.fh, jobs[7]["t_inactive"], 20, self.attrs
-        )
+        rpc_handle = flux.job.job_list_inactive(self.fh, jobs[7]["t_inactive"], 20)
 
         jobs_inactive = self.getJobs(rpc_handle)
 
         self.assertEqual(len(jobs_inactive), 7)
 
     # flux job list-inactive with name filter
-    def test_10_name_filter(self):
+    def test_15_list_inactive_name_filter(self):
         # submit a bundle of hostname jobs
         for i in range(5):
             jobid = self.submitJob(["hostname"])
 
         # 16 = 5 + 11 in previous tests
         self.waitForConsistency(16)
 
-        rpc_handle = flux.job.job_list_inactive(self.fh, 0.0, 20, self.attrs, "sleep")
+        rpc_handle = flux.job.job_list_inactive(self.fh, 0.0, 20, name="sleep")
 
         jobs_inactive = self.getJobs(rpc_handle)
 
         self.assertEqual(len(jobs_inactive), 11)
 
-        rpc_handle = flux.job.job_list_inactive(
-            self.fh, 0.0, 20, self.attrs, "hostname"
-        )
+        rpc_handle = flux.job.job_list_inactive(self.fh, 0.0, 20, name="hostname")
 
         jobs_inactive = self.getJobs(rpc_handle)
 
         self.assertEqual(len(jobs_inactive), 5)
 
+    # flux job list-id works
+    def test_16_list_id(self):
+        jobid = self.submitJob(["hostname"])
+
+        self.waitForConsistency(17)
+
+        rpc_handle = flux.job.job_list_id(self.fh, jobid)
+
+        job = rpc_handle.get_job()
+        self.assertEqual(job["id"], jobid)
+        # numbert of attributes should be atleast 10
+        self.assertGreater(len(job), 10)
```

Comment:
At one point it might be good to test that all expected attributes are there? I'm thinking something like `self.assertLessEqual({"userid", "status", "starttime", "id", ...}, set(job.keys()))`

Reply:
note to myself, noticed a typo in the above "numbert"